cmd/commands/git-source.go

```diff
@@ -294,7 +301,139 @@ func createCronExamplePipeline(opts *gitSourceCronExampleOptions) error {
+					deleteIfExistsFromMap(containerArr, "metadata")
+					deleteIfExistsFromMap(containerArr, "outputs")
+				}
+				containerVals, container := nestedMapLookup(rec, "container", "name")
+				if container != nil && containerVals != nil {
+					deleteIfExistsFromMap(container, "name")
+					deleteIfExistsFromMap(container, "resources")
+				}
+			}
+		}
+	}
+
+	//Delete redunded fields from sensor
+	_, githup := nestedMapLookup(crd, "spec", "github", "push", "id")
+	if githup != nil {
+		deleteIfExistsFromMap(githup, "id")
+		deleteIfExistsFromMap(githup, "owner")
+		deleteIfExistsFromMap(githup, "repository")
+	}
+
+	//Delete redunded fields from sensor
+	_, triggers := nestedMapLookup(crd, "spec", "triggers")
+	if triggers != nil {
+		for _, value := range triggers["triggers"].([]interface{}) {
+			if rec, ok := value.(map[string]interface{}); ok {
+				containerArrVals, containerArr := nestedMapLookup(rec, "template")
+				if containerArr != nil && containerArrVals != nil {
+					deleteIfExistsFromMap(containerArr, "metadata")
+					deleteIfExistsFromMap(containerArr, "outputs")
+				}
+				_, container := nestedMapLookup(rec, "template", "argoWorkflow")
+				if container != nil {
+					deleteIfExistsFromMap(container, "name")
+					deleteIfExistsFromMap(container, "resources")
+				}
+				_, resource := nestedMapLookup(rec, "template", "argoWorkflow", "source", "resource", "status")
+				if resource != nil {
+					deleteIfExistsFromMap(resource, "status")
+				}
+				_, metadata := nestedMapLookup(rec, "template", "argoWorkflow", "source", "resource", "metadata", "creationTimestamp")
+				if metadata != nil {
+					deleteIfExistsFromMap(metadata, "creationTimestamp")
+				}
+			}
+		}
+
+	}
+	//Delete redunded fields from eventSource
+	_, targetPort := nestedMapLookup(crd, "spec", "service", "ports")
+	if targetPort != nil {
+		for _, value := range targetPort["ports"].([]interface{}) {
+			if rec, ok := value.(map[string]interface{}); ok {
+				_, targetPort := nestedMapLookup(rec, "targetPort")
+				if targetPort != nil {
+					deleteIfExistsFromMap(targetPort, "targetPort")
+				}
+			}
+		}
+	}
+
+	deleteIfExistsFromMap(crd, "status")
+	metadata := crd["metadata"].(obj)
+	deleteIfExistsFromMap(metadata, "creationTimestamp")
+
+	data, err = yaml.Marshal(crd)
+	if err != nil {
+		return fmt.Errorf("failed to marshal cleaned up eventSorce yaml file. Error: %w", err)
+	}
+	err = billyUtils.WriteFile(gsFs, filename, data, 0666)
+	if err != nil {
+		return fmt.Errorf("failed to write cleaned up eventSorce yaml file. Error: %w", err)
+	}
 	return nil
+
+}
+func deleteIfExistsFromMap(m map[string]interface{}, key string) {
+	_, ok := m[key]
+	if ok {
+		delete(m, key)
+	}
+}
+
+func nestedMapLookup(m map[string]interface{}, ks ...string) (rval interface{}, mm map[string]interface{}) {
+	var ok bool
+
+	if len(ks) == 0 {
+		return nil, nil
+	}
+	if rval, ok = m[ks[0]]; !ok {
+		return nil, nil
+	} else if len(ks) == 1 {
+		return rval, m
```

Comment:
Hmm not sure I fully understand the purpose of returning both `rval` and `m` ? shouldn't this method just return a single value which is a pointer to the requested target field (something like `lodash get()` ) ? or do we need both `rval` and `m` ?